### PR TITLE
Endpoint: Shifts with related collections

### PR DESCRIPTION
**Pull request by sebastianjonasson**
Draft. We can maybe find a better name for the endpoint.

### CODE REVIEW

**Comment by sebastianjonasson** [comment]
As I see it the implementation will need to:

    1. Fetch shifts in interval
    2. Fetch timereports in interval OR shift-id in shifts-ids
    3. Fetch inquiries in interval OR shift-id in shifts-ids
    4. Fetch shifts where shift-id in shift-ids (from timereports, inquiries)
    5. Fetch approvals where report-vid in report-vids
    6. Fetch availabilities in interval

where 2-4 fetches orphaned entities.

Not sure if orphans applies for inquiries.

File: spec/paths/shifts@related-collections.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+get:
```
Comment (jheander) [inline spec/paths/shifts@related-collections.yaml:1]: The path `shifts/related-collections` is in conflict with the pre-existing path `shifts/:id`.
 Not sure about a better endpoint URL though :sweat_smile: 
Maybe `api/interval/v1/shifts`?

File: spec/paths/shifts@related-collections.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      schema:
+        type: array
+        example: '["timereports", "approvals", "inquiries", "availabilities"]'
+        items:
+          type: string
+          enum:
+            - timereports
+            - inquiries
+            - approvals
+            - availabilities
```
Comment (jheander) [inline spec/paths/shifts@related-collections.yaml:27]: Other APIs have a standard format for interval queries that we already have a parser implemented for:
`interval[start]=...&interval[end]=...&interval[match]=...` See shifts index endpoint for example.

File: spec/paths/shifts@related-collections.yaml (hunk 1 above)
Comment (sebastianjonasson) [reply spec/paths/shifts@related-collections.yaml]: Put it under `/api/related/v1/shifts`. Not sure if that is better. But indicates that the  `/api/related/v1` endpoint fetches related collections.

**Pull request merged**